.github/workflows/build-and-test-app.yml

```diff
@@ -0,0 +1,77 @@
+name: build-and-test
+run-name: build app from the ${{ github.head_ref }} branch
+on:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: "15 4,5 * * *"
```

Comment:
You mentioned in the PR description next:

> Add [on.schedule](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule)  to allow triggering a workflow at a scheduled time. To understand cron syntax follow to provided

Could you explain why we need to run tests `on schedule`?

Reply:
The next sentence in this paragraph: 

> It's necessary for us to prevent cache from removing (see [eviction policy](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy)).

Reply:
I got the idea about the caching policy, but why do we need to run it so often?

> GitHub will remove any cache entries that have not been accessed in over 7 days.